beluga_ros/include/beluga_ros/ndt_ellipsoid.hpp

```diff
@@ -0,0 +1,146 @@
+// Copyright 2024 Ekumen, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef BELUGA_ROS_NDT_ELLIPSOID_HPP
+#define BELUGA_ROS_NDT_ELLIPSOID_HPP
+
+#include <beluga_ros/messages.hpp>
+
+/**
+ * \file
+ * \brief Utilities for NDT I/O over ROS interfaces.
+ */
+
+namespace beluga_ros {
+
+template <typename RealScalar>
+inline bool isApprox(RealScalar x, RealScalar y, RealScalar prec = Eigen::NumTraits<RealScalar>::dummy_precision()){
+  return std::abs(x-y) < prec;
+}
+
+inline bool use_mean_covariance(auto& eigenSolver, const auto& cell, auto& marker){
+  eigenSolver.compute(cell.covariance);
+  if(eigenSolver.info() != Eigen::Success){
+    return false;
+  }
+  
+  // Compute the rotation based on the covariance matrix
+  const auto eigenvectors = eigenSolver.eigenvectors().real();
+  const auto eigenvalues = eigenSolver.eigenvalues();
+
+  Eigen::Matrix3d rotationMatrix;
+  Eigen::Vector3d scalevector;
+  rotationMatrix << eigenvectors.col(0), eigenvectors.col(1), eigenvectors.col(2);
+  scalevector = Eigen::Vector3d{eigenvalues.x(), eigenvalues.y(), eigenvalues.z()};
+
+  // Permutation
+  if(!isApprox(rotationMatrix.determinant(), 1.0)) {
+    rotationMatrix << eigenvectors.col(1), eigenvectors.col(0), eigenvectors.col(2);
+    scalevector = Eigen::Vector3d{eigenvalues.y(), eigenvalues.x(), eigenvalues.z()};
+  }
+
+  if(!isApprox(rotationMatrix.determinant(), 1.0)) {
+    return false;
+  }
```

Comment:
@Fernando-Sanz nit: consider flipping the order and computing fewer times:
```suggestion
  if(!isApprox(std::abs(rotationMatrix.determinant()), 1.0)) {
    return false;
  }
  
  // Permute columns to ensure the rotation is in fact a rotation
  if (rotationMatrix.determinant() < 0) {
    rotationMatrix << eigenvectors.col(1), eigenvectors.col(0), eigenvectors.col(2);
    scalevector = Eigen::Vector3d{eigenvalues.y(), eigenvalues.x(), eigenvalues.z()};
  }
```

Reply:
That way appear some cubes. A non-rotation matrix could have the determinant != -1 that change to 1 with the permutation, couldn't it?

Reply:
I think @hidmic intended the following:

```suggestion  
  // Permute columns to ensure the rotation is in fact a rotation
  if (rotationMatrix.determinant() < 0) {
    rotationMatrix << eigenvectors.col(1), eigenvectors.col(0), eigenvectors.col(2);
    scalevector = Eigen::Vector3d{eigenvalues.y(), eigenvalues.x(), eigenvalues.z()};
  }
  if(!isApprox(std::abs(rotationMatrix.determinant()), 1.0)) {
    return false;
  }
```